scripts/promote-beta-experimental-opt-in.ts

```diff
@@ -14,6 +18,22 @@ void runPromoteJob(jobName, () => {
   return createVersionsUpdatePullRequest((currentVersions) => {
     const ampVersion = AMP_VERSION || currentVersions.nightly.slice(2);
 
+    // for scheduled promotions, check that the new version is a forward promote
+    if (!AMP_VERSION) {
+      if (
+        !isForwardPromote(ampVersion, [
+          currentVersions['beta-opt-in'],
+          currentVersions['beta-traffic'],
+          currentVersions.stable,
+          currentVersions.lts,
+        ])
+      ) {
+        throw new Error(
+          'The scheduled promotion is older than current versions. This is most likely due to a stale nightly branch.'
+        );
```

Comment:
1. Do we really want to raise an error? How about logging it and quitting gracefully? (the answer could still be to raise an error, I just want to understand this choice)
2. The error message repeats in every file - in that case just move it into the `isForwardPromote` function and rewrite it as `ensureForwardPromote` so it'll do the error throwing on its own

Reply:
1. We don't want to quit gracefully because on-duty will expect a promote PR. An error is not the best but it will let onduty know that there won't be a promote.
2. done